sdk/metrics/src/test/java/io/opentelemetry/sdk/metrics/SdkMeterTest.java

```diff
@@ -31,20 +31,21 @@ void testLongCounter() {
             .build();
     assertThat(longCounter).isNotNull();
 
-    assertThat(
-            sdkMeter
-                .counterBuilder("testLongCounter")
-                .setDescription("My very own counter")
-                .setUnit("metric tonnes")
-                .build())
-        .isSameAs(longCounter);
+    // Note: We no longer get the same instrument instance as these instances are lightweight
+    // objects backed by storage now.  Here we just make sure it doesn't throw to grab
+    // a second instance.
+    sdkMeter
+        .counterBuilder("testLongCounter")
+        .setDescription("My very own counter")
+        .setUnit("metric tonnes")
+        .build();
 
     assertThatThrownBy(() -> sdkMeter.counterBuilder("testLongCounter").build())
         .isInstanceOf(IllegalArgumentException.class)
-        .hasMessage("Instrument with same name and different descriptor already created.");
+        .hasMessageContaining("Metric with same name and different descriptor already created.");
```

Comment:
hmm. it's not a `Metric` that's being created here, but an `Instrument`. I think this message should be reverted to what it was previously.